apps/examples/wc-vanilla/main.js

```diff
@@ -17,3 +23,7 @@ document.querySelector('#app').innerHTML = `
 `;
 
 setupCounter(document.querySelector('#counter'));
+
+document.querySelector('#modal-trigger').addEventListener('click', () => {
+    document.querySelector('#modal').setAttribute('isOpen', true);
```

Comment:
Out of curiosity, did you try something like `document.querySelector('#modal').showModal();`?

Reply:
It won't work as `#modal` is a `pie-component` not a `dialog`.

<img width="524" alt="image" src="https://github.com/justeattakeaway/pie/assets/448801/148d0b02-065c-4047-9feb-0e5787ccee7d">

Reply:
Ah I see, we would probably have to define a `showModal` method on our component (which calls the inner function) to get this to work. Thanks.